### ISSUE TITLE
Handle terraform apply in 2i2c GCP cluster aka. pilot-hubs cluster

### ISSUE THREAD

**Issue by consideRatio** [2i2c-org/infrastructure#4655]
This showed up for me doing an unrelated bucket permission change as requested by James in https://2i2c.slack.com/archives/C028WU9PFBN/p1724269701095679.

```terraform
  # google_filestore_instance.homedirs["filestore"] must be replaced
-/+ resource "google_filestore_instance" "homedirs" {
      ~ create_time      = "2023-06-14T21:48:47.876073454Z" -> (known after apply)
      ~ effective_labels = {} -> (known after apply)
      + etag             = (known after apply)
      ~ id               = "projects/two-eye-two-see/locations/us-central1-b/instances/pilot-hubs-homedirs" -> (known after apply)
      - labels           = {} -> null
        name             = "pilot-hubs-homedirs"
      ~ terraform_labels = {} -> (known after apply)
      + zone             = (known after apply)
        # (5 unchanged attributes hidden)

      ~ file_shares {
            name          = "homes"
          - source_backup = "projects/two-eye-two-see/locations/us-central1/backups/test" -> null # forces replacement
            # (1 unchanged attribute hidden)
        }

      ~ networks {
          ~ ip_addresses      = [
              - "10.234.45.250",
            ] -> (known after apply)
          ~ reserved_ip_range = "10.234.45.248/29" -> (known after apply)
            # (3 unchanged attributes hidden)
        }
    }
```

**Comment by consideRatio** [comment]
The `source_backup` part being removed "forces replacement" is the crux. Its documented in https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/filestore_instance#source_backup as: 
![image](https://github.com/user-attachments/assets/4e63564f-5e47-4c97-8d6e-18f653e8110b)

**Comment by consideRatio** [comment]
```bash
# this command led to change, error remains
terraform refresh
```

```bash
# this command shows us `source_backup=null` under `file_shares` in the terraform state
terraform state show 'google_filestore_instance.homedirs["filestore"]'`

# google_filestore_instance.homedirs["filestore"]:
resource "google_filestore_instance" "homedirs" {
    create_time      = "2023-06-14T21:48:47.876073454Z"
    description      = null
    effective_labels = {}
    etag             = null
    id               = "projects/two-eye-two-see/locations/us-central1-b/instances/pilot-hubs-homedirs"
    kms_key_name     = null
    labels           = {}
    location         = "us-central1-b"
    name             = "pilot-hubs-homedirs"
    project          = "two-eye-two-see"
    terraform_labels = {}
    tier             = "BASIC_HDD"

    file_shares {
        capacity_gb   = 5120
        name          = "homes"
        source_backup = null
    }

    networks {
        connect_mode      = "DIRECT_PEERING"
        ip_addresses      = [
            "10.234.45.250",
        ]
        modes             = [
            "MODE_IPV4",
        ]
        network           = "default"
        reserved_ip_range = "10.234.45.248/29"
    }
}
```

**Comment by consideRatio** [comment]
Hmmm, I think this is a consequence of testing the backup recovery. When we did a backup recovery, we got `file_shares.source_backup` set to `projects/two-eye-two-see/locations/us-central1/backups/test`, and with that being managed by terraform - setting it back to null makes it require re-creation.

I think if we can make terraform not manage that field, it would help us handle this situation. As a workaround, we could also set it explicitly to `projects/two-eye-two-see/locations/us-central1/backups/test` and update it again if we do a recovery.
